### PR TITLE
[docs] Minor improvements to Gentle Walkthrough example

### PR DESCRIPTION
**Pull request by arunppsg**
## Why are these changes needed?

- Improved readability and prose
- Added missing links and syntax highlighting

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by aslonnie — COMMENTED** [review]
@jjyao could you get some one from core team to have a look at this?

**Review by jjyao — APPROVED** [review]
Core side change looks good to me. @matthewdeng can you review the train related chnages?

**Comment by anyscalesam** [comment]
reminder @matthewdeng

File: doc/source/ray-core/examples/gentle_walkthrough.ipynb [hunk 1]
```diff
@@ -263,7 +249,7 @@
    "id": "c5dc2e18",
    "metadata": {},
    "source": [
-    "To retrieve database entries and measure performance, you do not need to make many changes to the code. Here's an overview of the process:"
+    "Here's an overview of the process to retrieve database entries and measure performance:"
```
Comment (angelinalg) [inline doc/source/ray-core/examples/gentle_walkthrough.ipynb:252]:
```suggestion
    "The following is an overview of the process to retrieve database entries and measure performance:"
```

File: doc/source/ray-core/examples/gentle_walkthrough.ipynb [hunk 2]
```diff
@@ -310,14 +296,14 @@
    "id": "8df4087c",
    "metadata": {},
    "source": [
-    "Running the task in parallel requires two minor code modifications.\n",
-    "To execute your Ray task remotely, you must use a `.remote()` call.\n",
+    "Running the task in parallel required two minor code modifications.\n",
+    "The first one is adding `ray.remote` decorator for making the function `retrieve_task` as a Ray task and the second one is using a `.remote()` call to execute your Ray task remotely.\n",
```
Comment (angelinalg) [inline doc/source/ray-core/examples/gentle_walkthrough.ipynb:300]:
```suggestion
    "First, add the `ray.remote` decorator to make the function `retrieve_task` a Ray task.  Second, use a `.remote()` call to execute your Ray task remotely.\n",
```

**Review by angelinalg — APPROVED** [review]
Thank you for improving Ray docs! Just had some nits to make edits more consistent with Anyscale writing style.

**Comment by arunppsg** [comment]
@angelinalg just a gentle reminder

**Comment by anyscalesam** [comment]
@arunppsg  can you fix the CI failures and we can attempt another merge?

**Pull request closed without merging**